lib/array.js

```diff
@@ -83,16 +117,21 @@ const asyncMap = (items, fn, options = {}, done) => {
 
   const next = () => {
     const itemsNumber = count ? countNumber() : min;
-    const iterMax = Math.min(items.length, itemsNumber + count);
+    const iterMax = Math.min(len, itemsNumber + count);
 
     begin = Date.now();
     for (; count < iterMax; count++) {
-      const itemResult = fn(items[count], count);
-      if (done) result[count] = itemResult;
+      const itemResult = fn(data.next().value, count);
+      if (done) {
+        if (name === 'Array') result.push(itemResult);
```

Comment:
FWIW, an object with these types as keys and an appropriate mapping functions as values is prettier for purpose like this. I suppose, that this piece of code will not be relevant after using `common.iter`, however just pointing it out for future reference.

Reply:
May I ask you a question?
If I replace my ```javascript const data = items[Symbol.iterator]();```
with ``` javascript const data = common.iter(items);``` I will get almost the same, and I have no need to rewrite code where I use ```data.next().value```.
So, Is it what I am expected to do?

Reply:
Yes, I've missconcepted this, you may need here to use a constructor name to add element to `result`. So it still make sense, AFAIS, to move this conditional statement to a separate function or to make an mapping object described previously, because it is used at least four times in this file, kind of boilerplate code, as for me.

Reply:
What do you think about this solution with separate function?
``` javascript
const push = (name, result, value) => {
  switch (name) {
    case 'Set':
      return result.add(value);
    case 'Map':
      return result.set(value[0], value[1]);
    case 'Array':
      return result.push(value);
    case 'String':
      return result += value;
  }
  return null;
}
```

Reply:
I prefer not using `switch-case` here.

Reply:
Transform to ```if/else``` construction?

Reply:
```javascript
const pushValue = {
  Array: (array, value) => array.push(value),
  Set: (set, value) => set.add(value),
  Map: (map, entry) => map.set(...entry),
  String: (string, value) => string += value,
};

const push = pushValue[name];
push(result, value);
```
Can also be smth like this. By the way, I don't see a point in returning `null` if no `name` was matched, because in this case an instance cannot be itterable and we can't execute a function, if I understand correctly. I think, this was noticed by @SemenchenkoVitaliy already.

Reply:
May I ask you again?
If I separate ```push function``` prettier code says that I should change my ```let result``` to ```const result``` How can I solute this? Can I turn off this option, or what?

Reply:
Also, If I give ```string``` into object pushValue, I won't concat my string with ```string += value```
If I do something like this, It will work ```  javascript  
if (name === 'String') result += value;
else push(name, result, value);
```

Reply:
I wonder, if you could simply use `common.Iterator.map()` and others methods and then just `common.Iterator.collectTo()`?

Reply:
May be, I will check.

Reply:
I suppose, this was actually meant by @o-rumiantsev